nondex-common/pom.xml

```diff
@@ -45,5 +45,35 @@
       </plugin>
     </plugins>
   </build>
-  
+
+  <profiles>
```

Comment:
So from my understanding, this profile is to ensure that if the underlying JVM is Java 9 or up, then the way to build the NonDex project should be configured differently? If NonDex is actually run on a project that uses Java 9+, would a NonDex built using Java 8 work on it?

Reply:
> So from my understanding, this profile is to ensure that if the underlying JVM is Java 9 or up, then the way to build the NonDex project should be configured differently? If NonDex is actually run on a project that uses Java 9+, would a NonDex built using Java 8 work on it?

Yes, you're correct. It is used to configure NonDex correctly to be able to build under Java 9 or up.

No, our current implementation requires the underlying JVM version to be consistent between `running project` and `NonDex` itself. Otherwise, there will be exceptions caused by incompatible version of instrumented classes. For E.g. if `NonDex` is built using Java 8 but the project is built with Java 11.
```
java/lang/UnsupportedClassVersionError: java/lang/Class has been compiled by a more recent version of the Java Runtime (class file version 55.0), this version of the Java Runtime only recognizes class file versions up to 52.0
```